Enabled nullable types in Tommy and added some safety checks.

emayevski opened the pull request.

JaggerJo commented:
btw. something is off with your file encoding. It seems like the line endings are changed which makes it impossible for git to diff it. 


(I manually fixed it by reformatting the code in your other PR)
<img width="1262" height="557" alt="image" src="https://github.com/user-attachments/assets/d0aa710a-bccf-4d32-be4c-c8d29997d215" />

emayevski commented:
> btw. something is off with your line endings. It seems like the line endings are changed which makes it impossible for git to diff it.

Locally, the Tommy.cs files have Windows EOLs. This is not something I changed explicitly. Do you (or Github) expect them to be Unix LFs? I can change EOLs and amend the commit (this will also update the PR).

JaggerJo commented:
> > btw. something is off with your line endings. It seems like the line endings are changed which makes it impossible for git to diff it.
> 
> Locally, the Tommy.cs files have Windows EOLs. This is not something I changed explicitly. Do you (or Github) expect them to be Unix LFs? I can change EOLs and amend the commit (this will also update the PR).

I just added a `.gitattributes` file that should normalize file encoding.

emayevski commented:
> > > btw. something is off with your line endings. It seems like the line endings are changed which makes it impossible for git to diff it.
> > 
> > 
> > Locally, the Tommy.cs files have Windows EOLs. This is not something I changed explicitly. Do you (or Github) expect them to be Unix LFs? I can change EOLs and amend the commit (this will also update the PR).
> 
> I just added a `.gitattributes` file that should normalize file encoding.

Thank you, I've made another commit (the second in this pull request), and hopefully, it will be correct.

JaggerJo commented:
Merged your changes into an intermediate branch. Had to run `git add --renormalize .`, but now I get a proper diff: https://github.com/Voyonic-Systems/Echoes/pull/10/files

emayevski commented:
Great. I will merge your changes back to my fork, and I hope that in the future, as the configuration file is in place, we won't have this formatting problem.
